core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/report/ReportEditViewModel.kt

```diff
@@ -0,0 +1,230 @@
+                hideBottomNavigation = false
+            )
+        }
+        launchIfHasPermission(
+            setLoadingState = true,
+            permissionCheck = { true }
+        ) {
+            async {
+                loadEntity(
+                    serializer = Report.serializer(),
+                    onLoadFromDb = { db ->
+                        db.reportDao().findByUid(entityUid)
+                    },
+                    makeDefault = {
+                        Report()
+                    },
+                    uiUpdate = { loadedReport ->
+                        _uiState.update { prev ->
+                            prev.copy(
+                                reportOptions2 = loadedReport?.reportOptions?.takeIf { it.isNotBlank() }
+                                    ?.let { Json.decodeFromString(ReportOptions2.serializer(), it) }
+                                    ?: ReportOptions2()
+                            )
+                        }
+                    }
+                )
+                navResultReturner.filteredResultFlowForKey(RESULT_KEY_REPORT).collect { result ->
+                    val reportResult = result.result as? Map<*, *> ?: return@collect
+                    val filter = reportResult["filter"] as? ReportFilter3 ?: return@collect
+                    val seriesId = (reportResult["reportSeriesUid"] as? Int) ?: return@collect
+                    println("Report filter: ${filter}")
+                    onFilterChanged(filter, seriesId)
+                }
+            }
+        }
+
+        _appUiState.update { prev ->
+            prev.copy(
+                actionBarButtonState = ActionBarButtonUiState(
+                    visible = true,
+                    text = systemImpl.getString(MR.strings.done),
+                    onClick = this@ReportEditViewModel::onClickSave
+                )
+            )
+        }
+    }
+
+    fun onClickSave() {
+        viewModelScope.launch {
+            val currentReport = _uiState.value.reportOptions2
+            val report = Report(
+                reportTitle = currentReport.title,
+                reportOptions = Json.encodeToString(currentReport),
+            )
+            scheduleEntityCommitToSavedState(report, serializer = Report.serializer())
+            activeRepo.withDoorTransactionAsync {
+                try {
+                    if (entityUid == 0L) {
+                        activeRepo.reportDao().insert(report)
+                        println("Report options inserted successfully: ${report}")
+                    } else {
+                        activeRepo.reportDao().update(report)
+                        println("Report options updated successfully: ${report}")
+                    }
+                } catch (e: Exception) {
+                    println("Error updating report options: ${e.message}")
+                }
+            }
+        }
+    }
+
+    fun onEntityChanged(newOptions: ReportOptions2) {
```

Comment:
onEntityChanged __must__ use ```scheduleEntityCommitToSavedState``` so that changes made by the user are saved to the saved state handle - which is required to ensure data is not lost when the app is destroyed / recreated and  on the web.